src/back/utils/validations.test.ts

```diff
@@ -21,3 +23,29 @@ describe('validateProposalId', () => {
     expect(() => validateProposalId('    ')).toThrow(RequestError)
   })
 })
+
+describe('validateEventTypesFilters', () => {
+  test('Should return an empty object when event_type is not provided', () => {
+    const req = { query: {} } as never
+    const result = validateEventTypesFilters(req)
+    expect(result).toEqual({})
+  })
+
+  test('Should convert event_type into an array when it is a string', () => {
+    const req = { query: { event_type: EventType.Voted } } as never
+    const result = validateEventTypesFilters(req)
+    expect(result).toEqual({ event_type: [EventType.Voted] })
+  })
+
+  test('Should keep event_type as an array when it is already an array', () => {
+    const req = { query: { event_type: [EventType.Voted, EventType.Commented] } } as never
+    const result = validateEventTypesFilters(req)
+    expect(result).toEqual({ event_type: [EventType.Voted, EventType.Commented] })
+  })
+
+  test('Should throw an error if EventFilterSchema returns an error', () => {
+    const req = { query: { event_type: 'single_event' } } as never
+
+    expect(() => validateEventTypesFilters(req)).toThrow()
+  })
+})
```

Comment:
does the `req` object always has a `query` key? (what if it doesn't? )

Reply:
By definition in the Request interface, it always has the query object